src/vivarium_gates_nutrition_optimization/components/observers.py

```diff
@@ -170,8 +170,15 @@ def setup(self, builder: Builder) -> None:
         for outcome in models.PREGNANCY_OUTCOMES:
             builder.results.register_observation(
                 name=f"pregnancy_outcome_{outcome}_count",
-                pop_filter=f'alive == "alive" and pregnancy_outcome == "{outcome}" and tracked == True',
-                requires_columns=["alive", "pregnancy_outcome"],
+                pop_filter=f'alive == "alive" and tracked == True '
+                f'and previous_pregnancy == "pregnant" and pregnancy == "parturition" '
+                f'and pregnancy_outcome == "{outcome}"',
```

Comment:
Flagging to confirm that this will capture outcomes even among mothers who die... I am not sure if this happens at the beginning or the end of the parturition state but will investigate and follow-up

Reply:
Death is the first thing that happens in a time-step. I believe all mortality due to pregnancy occurs in this time-step, so they will have died by the time we are observing them here.